Python/Product/Wsl/Debugger/WslLauncher.cs

```diff
@@ -0,0 +1,327 @@
+using Microsoft.PythonTools.Debugger.Remote;
+using Microsoft.PythonTools.Infrastructure;
+using Microsoft.PythonTools.Interpreter;
+using Microsoft.PythonTools.Project;
+using Microsoft.VisualStudio;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudioTools.Project;
+
+namespace Microsoft.PythonTools.Wsl.Debugger {
+    class WslPythonLauncher : IProjectLauncher {
+        private readonly IServiceProvider _serviceProvider;
+        private readonly LaunchConfiguration _config;
+
+        private static readonly Lazy<string> _bashPath = new Lazy<string>(() => {
+            var path = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.Windows), "SysNative", "bash.exe");
+            if (File.Exists(path)) {
+                return path;
+            }
+            path = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.System), "bash.exe");
+            if (File.Exists(path)) {
+                return path;
+            }
+            path = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.SystemX86), "bash.exe");
+            if (File.Exists(path)) {
+                return path;
+            }
+            return null;
+        });
+
+        public WslPythonLauncher(IServiceProvider provider, LaunchConfiguration config) {
+            _serviceProvider = provider;
+            _config = config;
+        }
+
+        public int LaunchProject(bool debug) {
+            return Launch(_config, debug);
+        }
+
+        public int LaunchFile(string/*!*/ file, bool debug) {
+            var config = _config.Clone();
+            config.ScriptName = file;
+            return Launch(config, debug);
+        }
+
+        private int Launch(LaunchConfiguration config, bool debug) {
+            if (debug) {
+                StartWithDebugger(config);
+            } else {
+                StartWithoutDebugger(config).Dispose();
+            }
+
+            return VSConstants.S_OK;
+        }
+
+        private string GetPyVariable(Version version) {
+            if (version == null) {
+                return "if hash python3 2>/dev/null; then export py=python3;" +
+                    "elif hash python2 2>/dev/null; then export py=python2;" +
+                    "elif hash python 2>/dev/null; then export py=python;" +
+                    "else echo 'Cannot locate Python'; read -p 'Press Enter to exit . . .'; exit 1;" +
+                    "fi";
+            }
+
+            return (
+                "if hash python{0}.{1} 2>/dev/null; then export py=python{0}.{1};" +
+                "elif hash python{0} 2>/dev/null; then export py=python{0}; echo 'WARNING: Using python{0} instead of python{0}.{1}';" +
+                "else echo 'Cannot locate Python {0}.{1}'; read -p 'Press Enter to exit . . .'; exit 1;" +
+                "fi"
+                ).FormatInvariant(version.Major, version.Minor);
+        }
+
+        private string CreateScriptFile(params string[] lines) {
```

Comment:
We're going to end up with double quoting then, which is also going to mess with user-provided script arguments. I'm not sure the benefit is actually worth it. Especially since right now I'm just trying to dump the code somewhere safe and get back to bugfixing ;)